apps/meteor/client/components/message/toolbar/useNewDiscussionMessageAction.tsx

```diff
@@ -1,35 +1,35 @@
-import { Meteor } from 'meteor/meteor';
-import { Tracker } from 'meteor/tracker';
+import { useSetModal, useSetting } from '@rocket.chat/ui-contexts';
+import React, { useEffect } from 'react';
 
-import CreateDiscussion from '../../../client/components/CreateDiscussion/CreateDiscussion';
-import { imperativeModal } from '../../../client/lib/imperativeModal';
-import { roomCoordinator } from '../../../client/lib/rooms/roomCoordinator';
-import { hasPermission } from '../../authorization/client';
-import { settings } from '../../settings/client';
-import { MessageAction } from '../../ui-utils/client';
+import { hasPermission } from '../../../../app/authorization/client';
+import { MessageAction } from '../../../../app/ui-utils/client/lib/MessageAction';
+import { roomCoordinator } from '../../../lib/rooms/roomCoordinator';
+import CreateDiscussion from '../../CreateDiscussion';
 
-Meteor.startup(() => {
-	Tracker.autorun(() => {
-		if (!settings.get('Discussion_enabled')) {
+export const useNewDiscussionMessageAction = () => {
+	const enabled = useSetting('Discussion_enabled', false);
+
+	const setModal = useSetModal();
+
+	useEffect(() => {
```

Comment:
Might be a good idea to also remove the button (`MessageAction.removeButton`) in this effect's clean up function.

Reply:
good catch, done!